pyproject.toml

```diff
@@ -9,7 +9,7 @@ target-version = ['py38', 'py39', 'py310', 'py311']
 [tool.ruff]
 line-length = 105 # more lenient than black due to long function signatures
 src = ["rustworkx", "setup.py", "retworkx", "tests"]
-select = [
+lint.select = [
```

Comment:
Is this really necessary? I don't see the advantage of doing this. We only use ruff for linting so it is implicit

Reply:
Ruff issued a warning asking me to do it when I ran the job (same with the other change to a table name) - I think they were planning to deprecate top-level select maybe?

Reply:
For context, since I'm sat at my computer now, here's what the tox job says without the two extra `lint.` bits, from a fresh environment rebuild:
```
lint: remove tox env folder /Users/jake/code/retworkx/.tox/lint
lint: install_deps> python -I -m pip install -c/Users/jake/code/retworkx/constraints.txt -U 'black~=22.0' ruff
lint: commands[0] /Users/jake/code/retworkx/tests> black --check --diff ../rustworkx ../tests ../retworkx
All done! ✨ 🍰 ✨
160 files would be left unchanged.
lint: commands[1] /Users/jake/code/retworkx/tests> ruff check ../rustworkx ../retworkx . ../setup.py
warning: The top-level linter settings are deprecated in favour of their counterparts in the `lint` section. Please update the following options in `/Users/jake/code/retworkx/pyproject.toml`:
  - 'select' -> 'lint.select'
  - 'per-file-ignores' -> 'lint.per-file-ignores'
lint: commands[2] /Users/jake/code/retworkx/tests> cargo fmt --all -- --check
lint: commands[3] /Users/jake/code/retworkx/tests> python /Users/jake/code/retworkx/tools/find_stray_release_notes.py
  lint: OK (16.18=setup[5.48]+cmd[4.62,0.56,5.17,0.36] seconds)
  congratulations :) (16.42 seconds)
```

Reply:
I guess they decided to change the name